### DISCUSSION TITLE
When is it ok to reuse a key ?

### DISCUSSION THREAD

**Discussion by mchagneux** [jax-ml/jax#10376] [Q&A] [answered]
Hi all, 
I have a practical question wrt PRNG states. Is it ok to reuse the same key when there is another source of randomness ? For example, I frequently do some sort of resampling by repeatedly calling

`draws = random.choice(key, a=a, p=p)`

where at least one of `a` or `p` has changed between each call (often due to some other form of randomness which involves other keys), and therefore I know that the draws are different at each call. In this case, is it ok to reuse the same key for all draws ? 

Thanks in advance

**Comment by YouJiacheng** [comment, marked as answer]
https://github.com/google/jax/blob/88ac4edf63625db10f92d5368a598205aac1deb4/jax/_src/random.py#L496-L499
In your case, I think you should not reuse a key.
Even if `p` has proper randomness(i.e. uniform in simplex), fix key still cause bias w.r.t. position(index).
If `a` has proper randomness(i.e. uniform in permutation), it seems ok. However, are you sure that `a` has such randomness?

**Reply by mchagneux** [reply]
Ok thank you for the details!
